v0/destinations/userlist/transform.js

```diff
@@ -0,0 +1,52 @@
+const _ = require("lodash");
+
+const { EventType } = require("../../../constants");
+const {
+  defaultRequestConfig,
+  defaultPostRequestConfig
+} = require("../../util");
+
+const { endpoint } = require("./config");
+
+function isBlank(value) {
+  return _.isEmpty(_.toString(value));
+}
+
+function buildResponse(message, destination) {
+  const response = defaultRequestConfig();
+  const { pushKey } = destination.Config;
+
+  response.endpoint = endpoint;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json",
+    Authorization: `Push ${pushKey}`
+  };
+  response.body.JSON = message;
+  response.statusCode = 200;
+
+  return response;
+}
+
+function processSingleMessage(message, destination) {
+  const messageType = message.type.toLowerCase();
+
+  if (isBlank(message.userId)) {
```

Comment:
We support passing the `userId` in multiple places. Check [here](https://github.com/rudderlabs/rudder-transformer/blob/ca66ab93ea4c9b41b3d7d66b2c5711f9747a2e41/v0/util/data/GenericFieldMapping.json#L41)

Please use the method `getFieldValueFromMessage` from [here](https://github.com/rudderlabs/rudder-transformer/blob/ca66ab93ea4c9b41b3d7d66b2c5711f9747a2e41/v0/util/index.js#L520) to check for a valid value of the `userId`

On a different note: does it make sense to use `anonymousId` instead of `userId` here?

Reply:
We explicitly don't process messages from unauthenticated users, so this is intentional.

Reply:
👍